eip-0020.md

```diff
@@ -0,0 +1,202 @@
+
+Below we describe the data formats specified by this EIP and refer to the
+formats defined in EIP-0019.
+
+Depending on the bytes size of `ReducedTransaction` it may not fit into 4K limit
+of QR codes. In this case or for whatever reason the dApp may encode the
+following data in the QR code on the payment screen or link to it:
+
+`ergopayurl://<URL without https>`, for example
+`ergopayurl://sigmausd.io/signingRequest/2001-16b8-66c4-b800-6e52-8ce4`
+
+When the Wallet application parses the QR code bytes it can use prefix URI scheme
+to learn the message type.
+
+The wallet application should request URL and obtain the following data (json format)
+
+```
+ErgoPaySigningRequest:
+  - transaction: ReducedTransaction
+  - description: String (optional)
+  - replyToUrl: String (optional)
+```
+
+When dapp does not make use of an extra request, the `ErgoPaySigningRequest` could 
+also be encoded in the QR code or link:
+
+`ergopay:<ReducedTransaction, base 64 url safe encoded>`
+
+Once the transaction details obtained, they CAN BE OPTIONALLY shown on the payment
+screen of the Hot Wallet application.
+
+After signing is performed and the `SignedTransaction` data is obtained, the 
+wallet app need to POST the following data to the dApp using `replyToUrl` from the
+signing request (json format).
+
+```
+ErgoPayTransactionId:
+  - txId: String
+```
+
+After `ErgoPayTransactionId` is sent, only then the dApp can submit the
+SignedTransaction to the blockchain. The order is important, otherwise, if
+transaction is included into blockchain and dApp notification fails the dApp
+will be out of sync with blockchain.
+
+## Implementation in Mobile Wallet
+TODO
+
+## Implementation in dApp
+TODO
+
+## Benefits for dApps
+Ergo Pay provides a fast, easy, and secure way for users to buy goods and
+services in a dApp or on a website. When supported, Ergo Pay can substantially
+increase checkout conversion rates, user loyalty and purchase frequency, and
+reduced checkout time. 
+
+- dApp or website don’t need to handle user's secrets (mnemonic/private keys).
+Instead, once the user has signed the transaction to confirm purchase intent,
+your app or website receives a transaction id to monitor payment status on the
+blockchain.
+- dApp's users don't need to worry about security of their private keys as the
+mobile wallet guarantees they never leave the mobile device
+- ErgoPay EIP is compatible with Cold Wallet EIP, thus users can use Cold Wallet
+devices to sign transaction within ErgoPay signing process.
+- adding Ergo Pay to product detail pages, the cart, checkout page, in payment
+settings, or anywhere else a user can choose ErgoPay as the payment method or
+initiate a purchase.
+- The payment screen can be presented immediately after the user taps the Ergo
+Pay button, without any interim screens or pop-ups except to prompt for
+necessary product details, such as size or quantity.
+- Ergo Pay is simple and universal. It supports all smart contracts and offers
```

Comment:
Technically, I don't think it supports "all smart contracts" but most of them. For example, it can not support multi-sig contracts.

Reply:
Ok, I think clarification is needed here. Do you mean it supports any smart contract in the outputs or inputs?
The way I understand `ReducedTransaction` is that it reduces the script to a simple operation like `proveDlog` that any wallet supports, so I think you mean inputs here.

Reply:
ReducedTransaction contains inputs reduced to SigmaBoolean, i.e. any sigma protocol proposition including multi-signatures. So, the claim is correct, it is indeed universal solution.

Reply:
Wow, this is exciting then.